packages/next/src/lib/metadata/resolvers/resolve-url.test.ts

```diff
@@ -106,3 +110,71 @@ describe('resolveAbsoluteUrlWithPathname', () => {
     })
   })
 })
+
+describe('getSocialImageFallbackMetadataBase', () => {
+  describe('fallbackMetadataBase', () => {
+    let originalEnv: NodeJS.ProcessEnv
+    function getSocialImageFallbackMetadataBaseHelper(): string {
+      return getSocialImageFallbackMetadataBase(null).fallbackMetadataBase.href
+    }
+
+    beforeEach(() => {
+      originalEnv = process.env
+    })
+
+    afterEach(() => {
+      delete process.env.VERCEL_URL
+      delete process.env.VERCEL_ENV
+      delete process.env.VERCEL_BRANCH_URL
+      delete process.env.VERCEL_PROJECT_PRODUCTION_URL
+
+      process.env = originalEnv
+    })
+
+    it('should return localhost url in local dev mode', () => {
+      // @ts-expect-error override process env
+      process.env.NODE_ENV = 'development'
+      expect(getSocialImageFallbackMetadataBaseHelper()).toBe(
+        'http://localhost:3000/'
+      )
+    })
+
+    it('should return metadataBase ', () => {
+      // @ts-expect-error override process env
+      process.env.NODE_ENV = 'production'
+      expect(getSocialImageFallbackMetadataBaseHelper()).toBe(
+        'http://localhost:3000/'
```

Comment:
Nit pick: this looks like the same value as dev so perhaps metadataBase can be mocked to see if the value is observed

Reply:
I renamed the test since these are for testing when `metadataBase` is not present. The present cases are covered with e2e tests